test/framework/clusterctl/clusterctl_helpers.go

```diff
@@ -203,9 +204,16 @@ func UpgradeManagementClusterAndWait(ctx context.Context, input UpgradeManagemen
 	}
 }
 
+type WorkloadClusterTemplateInput struct {
+	YAML        []byte
+	ClusterName string
+	Namespace   string
+}
+
 // ApplyClusterTemplateAndWaitInput is the input type for ApplyClusterTemplateAndWait.
 type ApplyClusterTemplateAndWaitInput struct {
 	ClusterProxy                 framework.ClusterProxy
+	WorkloadClusterTemplate      *WorkloadClusterTemplateInput
```

Comment:
Let's add some godoc how setting this field impacts the behavior of the util func

Reply:
I think I would prefer to have two functions, `ApplyClusterTemplateAndWait` (the current one) and `ApplyCustomClusterTemplateAndWait` (a new current one) instead of having one full of `if input.ConfigCluster.ControlPlaneMachineCount == nil`.
The code for applying the template can be refactored in a private func called by both

Reply:
Sounds fine to me